Client/Frontend/Browser/BrowserLocationView.swift

```diff
@@ -267,8 +274,25 @@ class BrowserLocationView : UIView, ToolbarTextFieldDelegate {
         return true
     }
 
-    func textFieldDidLongPress(textField: ToolbarTextField) {
-        delegate?.browserLocationViewDidLongPressLocation(self)
+
+    func longPress(gestureRecognizer: UIGestureRecognizer) {
+        if gestureRecognizer.state == .Began {
+            delegate?.browserLocationViewDidLongPressLocation(self, completion: { success in
```

Comment:
A completion handler in a delegate method is odd. Is that really needed?

What happens when `success` is false? Will `isHandlingLongPress` then stay `true`?

Can't you cancel the gesture instead of keeping that state? Maybe you can get rid of `shouldRecognizeSimultaneouslyWithGestureRecognizer` too then?

I'm just worried that this is fragile because there is so much extra state now.